scripts/perf_report.py

```diff
@@ -56,26 +80,46 @@ def do_main(args:list):
                 (nclients_field, nops_field, nthreads_field, svr_metric, cli_metric, thread_metric) = parse_perf_line_names(line)
                 heading_row = [ 'Run-Type', svr_metric, cli_metric, thread_metric ]
 
-            (run_type, svr_value, svr_str, cli_value, cli_str, thread_str) = \
+            (run_type, nthreads_field, svr_value, svr_str, cli_value, cli_str, thread_str) = \
                 parse_perf_line_values(line)
 
+            # Create hash-entry for baseline server/client metrics for this thread-count
             if run_type.startswith('Baseline'):
-                base_svr_value = svr_value
-                base_cli_value = cli_value
-
-            metrics_by_run[run_type] = [svr_value, cli_value, svr_str, cli_str, thread_str ]
-
+                # svr_value - baseline server-throughput
+                # cli_value - baseline client-throughput
+                baseline_by_threads[nthreads_field] = baseline_tuple(svr_value, cli_value)
+
+            # DEBUG: print(f"{run_type}, {thread_str}")
+
+            # ------------------------------------------------------------------
+            # We want to distribute the metrics for each run-type across the
+            # map of metrics arranged by #-of-server-threads.
+            # - If this thread-count is a new one, add the dictionary-entry
+            #   { run_type: [ <metrics> ] } to the metrics tracked per thread.
+            # - Otherwise, update the metrics previously seen for previous
+            #   run-types for this thread-count with the metrics of the
+            #   new run-type seen.
+            #
+            # This way, we will be rearranging the one-line metrics rows
+            # into a dictionary of dictionaries, arranged by thread-count.
+            # ------------------------------------------------------------------
+            by_nthreads_dict = OrderedDict()
+            if nthreads_field not in metrics_by_threads:
+                metrics_by_threads[nthreads_field] = { run_type: [svr_value, cli_value, svr_str, cli_str, thread_str ] }
+            else:
+                by_nthreads_dict = metrics_by_threads[nthreads_field]
+                by_nthreads_dict.update( {run_type: [svr_value, cli_value, svr_str, cli_str, thread_str ] } )
+                metrics_by_threads[nthreads_field] = by_nthreads_dict
```

Comment:
Major rework is in this block, to rearrange the metrics generated for each run-type to distribute them across a dictionary keyed on "# of server threads."

See the sample outputs posted in discussion item #79 .